beluga-mqtt/src/lib.rs

```diff
@@ -232,6 +261,55 @@ impl MqttClient {
         Ok(Subscriber(ctx.manager.subscribe_many(new_topics)))
     }
 
+    /// Subscribes to multiple MQTT topics with the specified Quality of Service
+    /// [QoS] level and returns an [OwnedSubscriber] that can be used to receive
+    /// published messages.
+    ///
+    /// # Arguments
+    /// - `topics`: An iterator of topic strings to subscribe to.
+    /// - `qos`: The Quality of Service level for the subscriptions.
+    ///
+    /// # Returns
+    /// A [Result] containing an [OwnedSubscriber] if the subscriptions are
+    /// successful, otherwise an [Error].
+    pub async fn subscribe_many_owned<Iter>(
```

Comment:
Is my understanding correct that if we lose connection to the broker the subscription stream ends and we won't re-subscribe on reconnect? This behaviour should be documented.

Reply:
`Beluga` subscriber only manages his state, the broker could resubscribe, but it doesn't impact the Beluga subscriber.
The purpose of [OwnedSubscriber](https://github.com/inomotech-foss/beluga/pull/49/files#diff-36a9697afddec4cc64fbb38902b283d1eabd1c66d5b9ca0d3278ce8113ddf77aR57) is to unsubscribe from the topic if it goes out of scope.

Reply:
I just think that beluga adds a lot of nice abstractions on top of the underlying mqtt client, so it's kind of unexpected to me that it doesn't handle re-subscribing after disconnects.
In a vehicle temporary disconnects are quite frequent and generally the broker isn't going to restore your subscriptions.

I think at some point we should implement the logic for this in beluga, but for now let's at least document it.